site/source/tutorials/005_form_widgets/index.md

```diff
@@ -58,107 +77,128 @@ Let's use those classes and a few virtual DOM nodes, to add the visual elements
 
-The first thing that we see is a `key` property. As mentioned before, a key is necessary whenever the same type of widget or virtual DOM element will be rendered by a widget. The `key` property is followed by the `invalid` property. Setting the `invalid` property to `true` will update the widget's appearance to indicate that the widget contains invalid data. The `label`, `placeholder`, and `type` fields map to their expected properties.
+The first thing that we see is a `key` property. As mentioned before, a key is necessary whenever the same type of widget or virtual DOM element will be rendered by a widget. The `label`, `placeholder`, and `required` fields map to their expected properties.
 
 The `value` property renders the value of the field that is passed into the widget via its properties. Notice that there is no code that manipulates this value within the widget. As parts of a [reactive framework](https://en.wikipedia.org/wiki/Reactive_programming), Dojo 2 widgets do not normally update their own state. Rather, they inform their parent that a change has occurred via events or some other mechanism. The parent will then pass updated properties back into the widget after all of the changes have been processed. This allows Dojo 2 applications to centralize data and keep the entire application synchronized.
 
-Finally, the `onChange` and `onBlur` properties are populated with [arrow functions](https://www.typescriptlang.org/docs/handbook/functions.html) that call the corresponding `WorkerForm` event handler. This is another common pattern within Dojo 2 applications - the `WorkerForm` does not expose any of the components that it is using to build the form. Rather, the `WorkerForm` manages its children internally and, if necessary, raises events to inform its parent of any changes. This decouples the consumers of the `WorkerForm` widget and frees them from having to understand the internal structure of the widget. Additionally, it allows the `WorkerForm` to change its implementation without affecting its parent as long as it continues to fulfill the `WorkerFormProperties` interface.
+The final property assigns the `onFirstNameInput` method to the `onInput` property. The `onFirstNameInput` method, in turn calls the `onFormInput` property, informing the `WorkerForm`'s parent that a change has occurred. This is another common pattern within Dojo 2 applications - the `WorkerForm` does not expose any of the components that it is using to build the form. Rather, the `WorkerForm` manages its children internally and, if necessary, calls its event properties to inform its parent of any changes. This decouples the consumers of the `WorkerForm` widget and frees them from having to understand the internal structure of the widget. Additionally, it allows the `WorkerForm` to change its implementation without affecting its parent as long as it continues to fulfill the `WorkerFormProperties` interface.
```

Comment:
"and frees them from having to understand the internal structure of the widget." -> frees them from having to feels odd to me, not sure of a better way to say that we're not introducing tight coupling or dependencies?